src/api/chat.ts

```diff
@@ -0,0 +1,48 @@
+import firebase from "firebase/app";
+
+import { VenueChatMessage, PrivateChatMessage } from "types/chat";
+
+export type SendVenueMessageProps = {
+  venueId: string;
+  message: VenueChatMessage;
+};
+
+export const sendVenueMessage = async ({
+  venueId,
+  message,
+}: SendVenueMessageProps) =>
+  await firebase
+    .firestore()
+    .collection("venues")
+    .doc(venueId)
+    .collection("chats")
+    .add(message);
+
+export const sendPrivateMessage = async (message: PrivateChatMessage) => {
+  // @debt This is the legacy way of saving private messages. Would be nice to have it saved in one operation
+  for (const messageUser of [message.from, message.to]) {
+    await firebase
+      .firestore()
+      .collection("privatechats")
+      .doc(messageUser)
+      .collection("chats")
+      .add(message);
+  }
```

Comment:
As noted, this is `legacy` way of saving the private message both  to the author and recipient `privateMessages` collection

Reply:
Yup. I was more meaning the specific syntax of the `for`/`of` used here.

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

Seemingly will run the code for each element in the array. Any reason we are doing it this way, instead of say `[].forEach()`? (not a big deal either way, but `forEach` seems more consistent with other locations in the code i've seen/remember so far.

```suggestion
  // @debt This is the legacy way of saving private messages. Would be nice to have it saved in one operation
  [message.from, message.to].forEach((messageUser) => {
    await firebase
      .firestore()
      .collection("privatechats")
      .doc(messageUser)
      .collection("chats")
      .add(message);
  })
```